### ISSUE TITLE
Misinterpretation of <Alt-...> key mappings on non-US (Finnish) keyboard layout

### ISSUE THREAD

**Issue by unclechu** [neovide/neovide#1901]
This seems like a leftover of https://github.com/neovide/neovide/issues/445

**Describe the bug**

I have some mappings for maximizing & equalizing the window sizes. Vertically and horizontally:

``` vim
" windows size minimization/maximization/normalization
nn <A-=> :winc =<CR>
nn <A--> :winc _<CR>
nn <A-\> :winc \|<CR>
" for finnish layout
nn <A-/> :winc \|<CR>
```

On Finnish keyboard layout they don’t work.

**To Reproduce**

In Neovide:

1. On Finnish layout press `LAlt` + `LShift` + `0` (to trigger `Alt` + `=`)
2. Nothing happens, unlike in TUI where `:winc =<cr>` is triggered, as expected

1. On Finnish layout press `LAlt` + `-` (on US layout it’s `/` key, to trigger `Alt` + `-`)
2. Instead of `winc _` `winc |` is triggered. It seems the keys are interpreted as if it was US layout, on US layout the same key is assigned to `/`, so it seems `<A-/>` is triggered instead. In TUI `winc |` is triggered, as expected.

1. On Finnish layout press `LAlt` + `LShift` + `7` (to trigger `Alt` + `/`)
2. Nothing happens, unlike in TUI where `:winc |` is triggered, as expected

**Expected behavior**

In Neovide:

1. On Finnish layout `LAlt` + `LShift` + `0` triggers `:winc =<cr>`, just like in TUI
2. On Finnish layout `LAlt` + `-` triggers `:winc |<cr>`, just like in TUI
3. On Finnish layout `LAlt` + `LShift` + `7` triggers `:winc |<cr>`, just like in TUI

**Desktop (please complete the following information):**

- OS: 23.05.997.ddf4688dc7a (Stoat)
- Neovide Version: `main` HEAD — f412399 (dated June 15)
- Neovim Version: `v0.9.1`

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

1. Pressing `<A-=>` on Finnish layout:
   - [.neovide-wrapped_rCURRENT.log](https://github.com/neovide/neovide/files/11781297/default.neovide-wrapped_rCURRENT.log)
   - [.neovide-wrapped__rCURRENT.log](https://github.com/neovide/neovide/files/11781299/default.neovide-wrapped__rCURRENT.log)
2. Pressing `<A-->` on Finnish layout:
   - [.neovide-wrapped_rCURRENT.log](https://github.com/neovide/neovide/files/11781313/default.neovide-wrapped_rCURRENT.log)
   - [.neovide-wrapped__rCURRENT.log](https://github.com/neovide/neovide/files/11781314/default.neovide-wrapped__rCURRENT.log)
   - [.neovide-wrapped_r2023-06-18_14-25-47.log](https://github.com/neovide/neovide/files/11781320/default.neovide-wrapped_r2023-06-18_14-25-47.log)
3. Pressing `<A-/>` on Finnish layout:
   - [.neovide-wrapped_rCURRENT.log](https://github.com/neovide/neovide/files/11781349/default.neovide-wrapped_rCURRENT.log)
   - [.neovide-wrapped__rCURRENT.log](https://github.com/neovide/neovide/files/11781351/default.neovide-wrapped__rCURRENT.log)

**Comment by fredizzimo** [comment]
Could you try with https://github.com/neovide/neovide/pull/1899, I think it should fix all the issues. At least all those mappings work on my machine with Kde Plasma wayland and the Swedish keyboard layout, which is virtually identical to the Finnish one. Even `alt+altgr` mappings work so `map <A-\> :lua print("hi")<cr>` works. 

And previously I tested the Finnish international layout with the following mappings
`map <C-«> :lua print("hello")<cr>` and `map <C-A-«> :lua print("world")<cr>.` And it can distinguish between `ctrl+shift+altgr+4` and `ctrl+shift+alt+altgr+4`

**Comment by unclechu** [comment]
@fredizzimo 
> Could you try with https://github.com/neovide/neovide/pull/1899, I think it should fix all the issues.

Actually, I’m already compiling it to try. I’ll report if the fix helps.

**Comment by unclechu** [comment]
@fredizzimo while these `<A-...>` keys are working my common window navigation keys are not:

``` vim
nn <C-h>     :winc h<CR>
xn <C-h>     <Esc>:winc h<CR>
nn <C-j>     :winc j<CR>
xn <C-j>     <Esc>:winc j<CR>
nn <C-k>     :winc k<CR>
xn <C-k>     <Esc>:winc k<CR>
nn <C-l>     :winc l<CR>
xn <C-l>     <Esc>:winc l<CR>
```

And it does not matter what layout it is. They do not work on US layout either.

So while #1899 fixes this particular issue it introduces this another one.

**Comment by unclechu** [comment]
I tried the HEAD of the MR: 1050ebaa0acc349345716a584de70da0dece75e2

**Comment by fredizzimo** [comment]
I can repeat it here and will fix it. There's a double mapping it sends `<ctrl+^h>`

**Comment by fredizzimo** [comment]
It should be fixed in #1899 now

**Comment by fredizzimo** [comment]
This should be fixed by #1899, it would be great if you can test that.

**Comment by unclechu** [comment]
@fredizzimo already did, seems to work fine now, thanks for the fix! Tested against b14fb402aa9a9a137594606787755f337e5da22f

**Comment by falcucci** [comment]
I had the same problem and indeed is fixed after #1899
